estimagic/visualization/comparison_plot.py

```diff
@@ -0,0 +1,440 @@
+    candidates = array[array <= value]
+    if len(candidates) > 0:
+        idx = (np.abs(candidates - value)).argmin()
+        return candidates[idx]
+    else:
+        return np.nan
+
+
+def _find_next_upper(array, value):
+    # adapted from https://stackoverflow.com/a/2566508
+    candidates = array[array > value]
+    if len(candidates) > 0:
+        idx = (np.abs(candidates - value)).argmin()
+        return candidates[idx]
+    else:
+        return np.nan
+
+
+# ===========================================================================
+# PLOTTING FUNCTIONS
+# ===========================================================================
+
+
+def _create_comparison_plot_components(df, heights, lower, upper, rect_widths, width):
+    groups = heights.index.tolist()
+    source_dict = {k: {} for k in groups}
+    figure_dict = {k: {} for k in groups}
+    glyph_dict = {k: {} for k in groups}
+
+    for param_group in groups:
+        group_df = df[df["group"] == param_group]
+        param_names = sorted(group_df["full_name"].unique())
+        for param in param_names:
+            param_src = ColumnDataSource(group_df[group_df["full_name"] == param])
+            param_plot = figure(
+                title=param,
+                plot_height=int(heights.loc[param_group]),
+                plot_width=width,
+                tools="reset,save",
+                y_axis_location="right",
+                x_range=[lower.loc[param_group], upper.loc[param_group]],
+            )
+
+            point_glyph = param_plot.rect(
+                source=param_src,
+                x="binned_x",
+                y="dodge",
+                width=rect_widths.loc[param_group],
+                height=1,
+                color="color",
+                selection_color="color",
+                nonselection_color="color",
+                alpha=0.5,
+                selection_alpha=0.7,
+                nonselection_alpha=0.3,
+            )
+            _add_hover_tool(param_plot, point_glyph, df)
+
+            param_plot.hbar(
+                source=param_src,
+                y="dodge",
+                left="conf_int_lower",
+                right="conf_int_upper",
+                height=0.01,
+                alpha=0.0,
+                selection_alpha=0.7,
+                nonselection_alpha=0.0,
+                color="color",
+                selection_color="color",
+                nonselection_color="color",
+            )
+
+            _style_plot(param_plot, param, param_names, param_group)
+
+            figure_dict[param_group][param] = param_plot
+            source_dict[param_group][param] = param_src
+            glyph_dict[param_group][param] = point_glyph
+
+    return source_dict, figure_dict, glyph_dict
+
+
+def _add_hover_tool(plot, point_glyph, df):
+    top_cols = ["model", "full_name", "final_value", "model_class"]
+    if "conf_int_lower" in df.columns and "conf_int_upper" in df.columns:
```

Comment:
Quick and dumb question because I do not have the overview. Are the confidence columns not filled with nans if they do not exist? Thus, they are always there.

Reply:
At the moment at least not.

Reply:
Not even in ``_df_with_all_results(res_dict)``?

Reply:
You are right. Fixed in ec59c2a2b28e6c8928c82826fef5e2aa482b1a95.